src/osd/OSD.cc

```diff
@@ -2887,6 +2887,8 @@ will start to track new ops received afterwards.";
 
   else if (prefix == "heap") {
     ret = ceph::osd_cmds::heap(*cct, cmdmap, *f, ss);
+    outbl.append(ss);
```

Comment:
instead of moving the content from `ss` to `outbl` which is a little bit confusing. i think a better approach is to refactor `ceph::osd_cmds::heap()` so it returns a tuple of `<int, str, bufferlist>`.

Reply:
@tchaikov `ceph_heap_profiler_handle_command` produces output to stringstream. So any change in `ceph::osd_cmds::heap()` would repack it to bufferlist.
This change reaches its goal, and is short.

Reply:
i understand this change "reaches" its goal, and i agree it is short. but it's but confusing. i can help to refactor it later on with you permission.

Reply:
@tchaikov I added next commit for clarity. If you like it, I will squash them together.